ci/docker/debian-experimental-cpp.dockerfile

```diff
@@ -111,7 +111,7 @@ RUN /arrow/ci/scripts/install_sccache.sh unknown-linux-musl /usr/local/bin
 # Prioritize system packages and local installation.
 ENV ARROW_ACERO=ON \
     ARROW_AZURE=ON \
-    ARROW_BUILD_TESTS=ON \
+    ARROW_BUILD_TESTS=OFF \
```

Comment:
I'll revert this, but since we don't build tests on CRAN I'm checking to see if this being off gets us to a successful build.

Reply:
Could you try `BUILD_WARNING_LEVEL=PRODUCTION` instead?
It doesn't add `-Werror`.